layouts/shortcodes/apidocs.md

```diff
@@ -7,6 +7,10 @@
     {{ with $.Site.GetPage "/docs/languages/net/metrics-api" -}}
         {{ $pages = $pages | append (dict "lang" $value "page" .) }}
     {{ end }}
+    {{ else if eq $key "java" -}}
+    {{ with $.Site.GetPage "/docs/languages/java/learn-more/api" -}}
+        {{ $pages = $pages | append (dict "lang" $value "page" .) }}
+    {{ end }}
```

Comment:
Realized that without this, there's no java entry in the "Language APIs & SDKs -> API References" section. Moving the javadoc to be nested under "Learn more" means that we need to add a special case to this shortcode like dotnet.

On a separate note, while I've linked to the javadoc page for consistency with the other languages that have an external link, it would be more useful to link to the [Record Telemetry with API](http://localhost:1313/docs/languages/java/api-components/) page, since that includes explanations, examples, and javadoc links for all the key API surface area.

What we link to is a question of whether we value consistency or usefulness more.

![Screenshot 2024-11-05 at 8 55 55 AM](https://github.com/user-attachments/assets/5a0d204c-4033-42b3-813a-aa1845a9a0a5)

Reply:
the purpose of this list is to make the API references (read: JavaDoc) accessible more easily. So while I get your point that people should read the api components page first, this is more for people that have read that page and need quick access to JavaDocs (or the equivalent in another language)

I think this is the only thing that bothers me with this change, that we break that consistent like `/docs/languages/<language>/api` to point to the available documentation

I wonder if we could have a canonical URL **outside** the structure, like `/api/<language>`

cc @chalin

Reply:
> > I think this is the only thing that bothers me with this change, that we break that consistent like /docs/languages//api to point to the available documentation
> 
> To be fair, dot net already breaks that uniformity. I understand your point though.

Yes, but I would call that "for historical" reasons. We had a separation for JS in the past as well since the published their JsDocs for API & SDK on different places, but I think that's not the case anymore so we have only one link.

Reply:
What do you think about:

- Remove the external javadoc redirect page at `/docs/languages/java/learn-more/api`
- Rename the "Record Telemetry with API" from `/docs/languages/java/api-components` to `/docs/languages/java/api`. This page becomes Java's canonical API page.

If we do this, then the only inconsistency is that java's API link is internal rather than external.